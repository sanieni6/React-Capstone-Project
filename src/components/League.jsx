import React from 'react';
import PropTypes from 'prop-types';

const League = ({
  id, logos, name, handleClick,
}) => (

  <li key={id} className="league">
    <button className="league-btn" onClick={() => handleClick(id)} key={id} type="button">
      <img className="league-logo" src={logos.light} alt={name} />
      <p className="league-name">{name}</p>

    </button>

  </li>
);

League.propTypes = {
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  logos: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]).isRequired,
  name: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default League;
