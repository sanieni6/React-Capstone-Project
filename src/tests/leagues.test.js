import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import League from '../components/League';

describe('League component', () => {
  const leagueData = {
    id: 1,
    name: 'Premier League',
    logos: {
      light: 'https://example.com/premier-league-light.png',
      dark: 'https://example.com/premier-league-dark.png',
    },
    handleClick: jest.fn(),
  };

  test('renders correctly', () => {
    const { container } = render(<League
      id={leagueData.id}
      logos={leagueData.logos}
      name={leagueData.name}
      handleClick={leagueData.handleClick}
    />);
    expect(container).toMatchSnapshot();
  });

  test('handles click event', () => {
    const { getByRole } = render(<League
      id={leagueData.id}
      logos={leagueData.logos}
      name={leagueData.name}
      handleClick={leagueData.handleClick}
    />);
    getByRole('button').click();
    expect(leagueData.handleClick).toHaveBeenCalledWith(1);
  });

  test('displays league name and logo', () => {
    const { getByText, getByAltText } = render(<League
      id={leagueData.id}
      logos={leagueData.logos}
      name={leagueData.name}
      handleClick={leagueData.handleClick}
    />);
    expect(getByText('Premier League')).toBeInTheDocument();
    expect(getByAltText('Premier League')).toHaveAttribute('src', 'https://example.com/premier-league-light.png');
  });
});
