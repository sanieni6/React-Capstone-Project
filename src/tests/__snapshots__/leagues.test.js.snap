// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`League component renders correctly 1`] = `
<div>
  <li
    class="league"
  >
    <button
      class="league-btn"
      type="button"
    >
      <img
        alt="Premier League"
        class="league-logo"
        src="https://example.com/premier-league-light.png"
      />
      <p
        class="league-name"
      >
        Premier League
      </p>
    </button>
  </li>
</div>
`;
